ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/template2/Template2.kt

```diff
@@ -0,0 +1,160 @@
+import com.revenuecat.purchases.ui.revenuecatui.backgroundColor
+import com.revenuecat.purchases.ui.revenuecatui.callToActionBackgroundColor
+import com.revenuecat.purchases.ui.revenuecatui.callToActionForegroundColor
+import com.revenuecat.purchases.ui.revenuecatui.getActivity
+import com.revenuecat.purchases.ui.revenuecatui.getColors
+import com.revenuecat.purchases.ui.revenuecatui.localizedConfig
+import com.revenuecat.purchases.ui.revenuecatui.text1Color
+
+@Composable
+internal fun Template2(state: PaywallViewState.Template2, viewModel: PaywallViewModel) {
+    Column(
+        modifier = Modifier.fillMaxSize(),
+    ) {
+        Spacer(modifier = Modifier.weight(1f))
+        Template2MainContent(state, viewModel)
+        Spacer(modifier = Modifier.weight(1f))
+        PurchaseButton(state, viewModel)
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(
+                    start = UIConstant.defaultHorizontalPadding,
+                    end = UIConstant.defaultHorizontalPadding,
+                    bottom = UIConstant.defaultButtonVerticalSpacing,
+                ),
+            horizontalArrangement = Arrangement.Center,
+        ) {
+            RestorePurchasesButton(viewModel)
+        }
+    }
+}
+
+@Composable
+private fun Template2MainContent(state: PaywallViewState.Template2, viewModel: PaywallViewModel) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .verticalScroll(rememberScrollState())
+            .padding(horizontal = UIConstant.defaultHorizontalPadding),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(UIConstant.defaultButtonVerticalSpacing, Alignment.CenterVertically),
+    ) {
+        // TODO-PAYWALLS: Replace with correct image
+        val drawable = LocalContext.current.packageManager.getApplicationIcon(LocalContext.current.packageName)
+        Image(drawable.toBitmap(config = Bitmap.Config.ARGB_8888).asImageBitmap(), contentDescription = "")
+
+        val localizedConfig = state.paywallData.localizedConfig()
+        Text(
+            style = MaterialTheme.typography.displaySmall,
+            fontWeight = FontWeight.Black,
+            textAlign = TextAlign.Center,
+            text = localizedConfig?.title ?: "",
+            color = state.paywallData.getColors().text1Color,
+        )
+        Text(
+            style = MaterialTheme.typography.bodyLarge,
+            fontWeight = FontWeight.Bold,
+            textAlign = TextAlign.Center,
+            text = localizedConfig?.subtitle ?: "",
+            color = state.paywallData.getColors().text1Color,
+        )
+        state.packages.forEach { aPackage ->
+            SelectPackageButton(state, aPackage, viewModel)
+        }
+    }
+}
+
+@Composable
+private fun PurchaseButton(state: PaywallViewState.Template2, viewModel: PaywallViewModel) {
```

Comment:
Some of these can probably get extracted once we start working on other Templates.

Reply:
👍🏻 

Not necessary now but yeah definitely.

Reply:
Also per the suggested refactor, that state would become `PaywallViewState.Template` instead which makes more sense.